src/main/java/org/example/model/LineItem.java

```diff
@@ -0,0 +1,53 @@
+package org.example.model;
+
+import javax.persistence.*;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+@Entity
+@NoArgsConstructor
+@Getter
+public class LineItem {
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private Long id;
+
+  private Integer quantity;
+  private double totalPrice;
+  private String currency;
+
+  @ManyToOne
+  @JoinColumn(name = "order_id")
+  private Order order;
+
+  @OneToOne
+  @JoinColumn(name = "product_id", referencedColumnName = "id")
+  private Product product;
+
+  public LineItem(
```

Comment:
Use the [Currency class.](https://www.baeldung.com/java-money-and-currency)

Reply:
Done.